ha/util/health_generator/mock_health_event_publisher.py

```diff
@@ -0,0 +1,217 @@
+    Fetches disk ids using ConfStore search API and displays the result.
+    Args:
+    conf_store: ConftStoreSearch object
+    node_id: machine_id value
+    """
+    # TODO: This is temporary code till this is avalable in HA.
+    node_id = args.node_id
+    num_of_cvgs = Conf.get(_index, f'node>{node_id}>storage>num_cvg')
+    disk_ids = []
+    for cvg in range(num_of_cvgs):
+        num_of_data_disks = Conf.get(_index, f'node>{node_id}>storage>cvg[{cvg}]>devices>num_data')
+        num_of_metadata_disks = Conf.get(_index, f'node>{node_id}>storage>cvg[{cvg}]>devices>num_metadata')
+        [ disk_ids.append(Conf.get(_index, f'node>{node_id}>storage>cvg[{cvg}]>devices>data[{dt_disk}]')) for dt_disk in range(num_of_data_disks)]
+        [ disk_ids.append(Conf.get(_index, f'node>{node_id}>storage>cvg[{cvg}]>devices>metadata[{md_disk}]')) for md_disk in range(num_of_metadata_disks)]
+    print(disk_ids)
+
+def get_cvgs(args, conf_store: ConftStoreSearch = None) -> None:
+    """
+    Fetches cvg ids using ConfStore search API and displays the result.
+    Args:
+    conf_store: ConftStoreSearch object
+    node_id: machine_id value
+    """
+    node_id = args.node_id
+    # TODO: This is temporary code till this is avalable in HA.
+    cvg_count = Conf.get(_index, f'node>{node_id}>storage>num_cvg')
+    print([Conf.get(_index, f'node>{node_id}>storage>cvg[{cvg}]>name') for cvg in range(cvg_count)])
+
+def publish(args, conf_store: ConftStoreSearch = None) -> None:
+    """
+    publishes the message on the message bus.
+    Args:
+    conf_store: ConftStoreSearch object
+    node_id: machine_id value
+    config_file: config file path
+    """
+    try:
+        with open(args.file, 'r') as fi:
+            events_dict = json.load(fi)
+            if _events_key in events_dict.keys():
+                ConfigManager.init(None)
+                MessageBus.init()
+                message_type = Conf.get(const.HA_GLOBAL_INDEX, f'FAULT_TOLERANCE{const._DELIM}message_type')
+                message_producer = MessageBus.get_producer("health_event_generator", message_type)
+                cluster_id = Conf.get(const.HA_GLOBAL_INDEX, f'COMMON_CONFIG{const._DELIM}cluster_id')
+                site_id = Conf.get(const.HA_GLOBAL_INDEX, f'COMMON_CONFIG{const._DELIM}site_id')
+                rack_id = Conf.get(const.HA_GLOBAL_INDEX, f'COMMON_CONFIG{const._DELIM}rack_id')
+                storageset_id = '1' # TODO: Read from config when available.
+                for _, value in events_dict[_events_key].items():
+                    resource_type = value[_resource_type_key]
+                    if resource_type not in ['node', 'cvg', 'disk']:
+                        raise Exception(f'Invalid resource_type: {resource_type}')
+                    resource_status = value[_resource_status_key]
+                    if resource_status not in ['recovering', 'online', 'failed', 'unknown', 'degraded', 'repairing', 'repaired', 'rebalancing', 'offline']:
```

Comment:
Should we use constants defined in HA here and for resource types  ?

Reply:
Done, reading supported resource types from HA config file now.